OpenSim/Tools/IMUInverseKinematicsTool.cpp

```diff
@@ -164,16 +164,39 @@ void IMUInverseKinematicsTool::runInverseKinematicsWithOrientationsFromFile(
     ikSolver.setAccuracy(accuracy);
 
     auto& times = oRefs.getTimes();
-
+    Storage* modelOrientationErrors =
```

Comment:
Would be great to see this implemented using a `TableReporter`. IMHO we should be committed to the removal of Storage or at least not to propagate its usage.

Reply:
Thanks for the feedback @aseth1 👍 
The reason we can't use TableReporter here (or in older IKTool) is that Reporter requires "Outputs" to connect to its inputs. None of these quantities are outputs AFAICT. We could however use a TimeSeriesTable instead of Storage object and then write to storage file thus avoiding the use of Storage class.

Reply:
Markers error reporting has been there for a while though works differently by reporting only worst marker, are you proposing changing the behavior? or additional reporting? @aseth1

Reply:
Sorry for my brevity:
1. I  did mean using a `TableReporter` and to use `Outputs`.  The `Component` abstraction is not only for `ModelComponent`s but for Solvers and Operators and other calculators that use System resources. It would be straightforward to make `Solver` derive from `Component`. `InverseKinematicsSolver` would have `ListOutput` with a `Channel` for every error. These would include Coordrinate, Marker (distance) and Orientation (angle) errors. Then, we would not be restricted to accessing errors in files only, but using Solvers in the context of broader pipeline. That was always the intent.
2. I mean actual reporting of  Marker errors as quantities, not just the text of the worst marker in the Messages window (current behavior). Many users (including myself) have scripts that diff the marker data in a trc file against marker locations reported by IK in .sto so they can compute distances and use them to refine scaling and IK. With Outputs, it would be easier for users could to select just the Markers/Sensors/Coordinates they care about.

Since the functionality has not yet been exposed and widely used, I think now would be the best time to make the upgrade.

Reply:
Thanks for the feedback @aseth1, since this is a major conceptual change that needs further discussion (of whether Component is the best abstraction to derive Solvers from)which is way beyond the scope of this PR and the fact that local users are waiting on this for ongoing research. I opened issue #2740 to continue the discussion while not blocking ongoing research.